### PR TITLE
Add linz-bde-schema-publish script

### PR DESCRIPTION
**Pull request by strk**
As a facility, the schema will create and maintain
a "bde" publication for replication of tables under
the "bde" schema.

### CODE REVIEW

**Comment by strk** [comment]
I affirmed it would have been better to explicitly list all tables that need to go in the publication, but in its current state the schema loader does not even try to create the tables if the "bde" schema already exists, so it's harder to centralize such tables list.

File: sql/07-bde_replication.sql [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                AND c.relnamespace = n.oid
+                AND c.relname = 'pg_publication' )
+    THEN
+        -- This version of PostgreSQL
+        -- does not support logical replication
+        RETURN;
+    END IF;
+
+    -- NOTE: we're using EXECUTE here to skip run for PostgreSQL
+    --       not supporting 'PUBLICATION' syntax (< 10)
```
Comment (palmerj) [inline sql/07-bde_replication.sql]: Should we rather just check for > PG 10 in this function, or at least catch the exception?

Comment (strk) [reply sql/07-bde_replication.sql]: I've checked that PostgreSQL 9.6 does not have the pg_publication table, so the check should be equivalent and not require a version parsing routine. I don't expect any exception (tested against 9.6 and 10)

**Comment by strk** [comment]
I'm thinking it would be better to just have a `linz-bde-schema-publish` and a `linz-bde-schema-subscribe` pair of scripts, rather than building this into the schema loader itself.

**Comment by strk** [comment]
I've redone the PR to use a separate script `linz-bde-schema-publish`

**Comment by strk** [comment]
The script at the moment DROPS an existing publication, if any, and create a new one.
I'm afraid this could be destructive for existing subscribers that are way behind with history, in that the drop of publication might remove pending WAL files. I've to do more research about this.

**Comment by palmerj** [comment]
> The script at the moment DROPS an existing publication, if any, and create a new one.

Yup bad idea. Once it's created and in use it should only be altered

**Comment by strk** [comment]
With bd98a72 I removed the DROP PUBLICATION.
Still I'm not sure how to deal with publication groups, maybe the script could take a list (or read it from a file)

**Comment by palmerj** [comment]
> Still I'm not sure how to deal with publication groups, maybe the script could take a list (or read it from a file)

I think you need to discuss that with @imincik and update his digram to reflect the decision

File: scripts/linz-bde-schema-publish.in [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+if test -z "$DB_NAME"; then
+    echo "Usage: $0 { <database> | - }" >&2
+    echo "       $0 --version" >&2
+    exit 1
+fi
+
+export PGDATABASE=$DB_NAME
+
+# Configure environment for dbpatch-loader
```
Comment (imincik) [inline scripts/linz-bde-schema-publish.in]: * 'dbpatch-loader' - looks like copy/paste error

File: scripts/linz-bde-schema-publish.in [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+if test -n "$PGBIN"; then
+    PATH=$PATH:$PGBIN
+else
+    # Wild guess where table_version-loader and dbpatch-loader
+    # will be found (as of versions 1.4.0 and 1.2.0 they were
+    # installed in PostgreSQL bin dir)
+    for f in /usr/lib/postgresql/*/bin/; do
+      PATH="$PATH:$f";
+    done
+fi
```
Comment (imincik) [inline scripts/linz-bde-schema-publish.in]: * shouldn't be needed anymore as those scripts are installed in `/usr/bin` now. 
* also 'table_version-loader and dbpatch-loader' looks like copy/paste error

File: scripts/linz-bde-schema-publish.in [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    IF NOT EXISTS ( SELECT 1 FROM pg_catalog.pg_publication
+                    WHERE pubname = 'bde' )
+    THEN
+        CREATE PUBLICATION bde;
+    END IF;
+    ALTER PUBLICATION bde OWNER TO bde_dba;
+
+    FOR rec IN select c.relname from pg_class c, pg_namespace n
+        WHERE n.nspname = 'bde' and c.relnamespace = n.oid
+        AND c.relkind = 'r' AND c.relname NOT IN (
```
Comment (imincik) [inline scripts/linz-bde-schema-publish.in]: * isn't it better to use `information_schema.tables` instead lower level pg tables ?

**Comment by strk** [comment]
Thanks @imincik I've removed the offending lines (were unused)

**Pull request merged**